fe_utils/finite_elements.py

```diff
@@ -103,8 +210,17 @@ def tabulate(self, points, grad=False):
         <ex-tabulate>`.
 
         """
+        
+        vander_matrix = vandermonde_matrix(self.cell, self.degree, points, grad)
+        if grad:
+            return np.einsum("ilk,lj -> ijk", vander_matrix, self.basis_coefs)
+        
+        tabulation = vander_matrix @ self.basis_coefs
+        return tabulation
```

Comment:
There is a straightforward way of doing this in a single case. Look at how to use `...` in [`einsum`](https://numpy.org/doc/stable/reference/generated/numpy.einsum.html)